giskard/client/dtos.py

```diff
@@ -28,6 +35,66 @@ class TestSuiteDTO(ConfiguredBaseModel):
     function_inputs: List[TestInputDTO]
 
 
+class TestSuiteExecutionResult(str, Enum):
+    IN_PROGRESS = ("IN_PROGRESS",)
+    CANCELLED = ("CANCELLED",)
+    PASSED = ("PASSED",)
+    FAILED = ("FAILED",)
+    ERROR = "ERROR"
+
+
+class MLWorkerWSTestMessageType(str, Enum):
+    ERROR = ("ERROR",)
+    INFO = "INFO"
+
+
+class TestResultMessageDTO(ConfiguredBaseModel):
+    type: MLWorkerWSTestMessageType
+    text: str
+
+
+class SaveSuiteTestExecutionDetailsDTO(ConfiguredBaseModel):
+    inputs: Dict[str, List[str]]
+    outputs: List[str]
+    results: List[TestResultStatusEnum]
+    metadata: Dict[str, List[str]]
+
+
+class SaveSuiteTestExecutionDTO(ConfiguredBaseModel):
```

Comment:
That's a lot of data. Everything is needed ?

Reply:
Everything is saved into the Hub. 

I agree that we don't use nor need all those information into the Hub. We might probably want to cleanup everything. (Both from Hub and programatic execution)